### PR TITLE
JavaScript: Improve handling of re-exports in API graphs

### PR DESCRIPTION
**Pull request by max-schaefer**
Previously, API graphs didn't deal very well with re-exports. Specifically, if you had a package whose main `index.js` file was something like

```js
module.exports = require('./implementation.js')
```

and `implementation.js` had _individual_ exports (as opposed to a bulk export using `module.exports = ...`), those individual exports would not be reflected in the API graph of the package.

That's because we use type (back-)tracking to figure out where the value assigned to `module.exports` comes from, but if `implementation.js` doesn't use a bulk export then there isn't a node to step back to.

This would be fixed by introducing synthetic data-flow nodes representing the (initial) values of `exports` and `module`, as discussed previously, but it's a bit of a fiddly change, so this PR proposes to fix it at the API-graph level instead.

[Evaluation](https://github.com/max-schaefer/dist-compare-reports/blob/52c948de5a2f5286ac009b4e6b05023e83f34081/js/api-graph-reexports/report.md) (internal link) shows no performance changes (after the customary initial wobbles). API-graph metrics show significant numbers of new edges for a few databases; `azure-sdk-node`, `amphtml`, and `node`, in particular, see an increase of 30%, and `cornerstoneWADOImageLoader` quadruples its number of edges. However, there are no corresponding increases in runtimes or tuple counts, and for many databases there are no changes at all.

### CODE REVIEW

**Comment by max-schaefer** [comment]
This PR would benefit from the new `getAnExportedValue` predicate introduced in https://github.com/github/codeql/pull/4247. I'd be happy to wait until that PR is merged and then rebase.

File: javascript/ql/src/semmle/javascript/ApiGraphs.qll
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/ApiGraphs.qll:616]: Should there also be a case that steps to `getAPropertySource("exports")` to handle assignments to `module.exports`?

From the PR description it sounds like that case was already working, but just thought I'd check.

**Review by asgerf — COMMENTED** [review]
LGTM. The `getAnExportedValue` PR should land very soon, so if that would help let's wait for it.

File: javascript/ql/src/semmle/javascript/ApiGraphs.qll
Comment (max-schaefer) [reply javascript/ql/src/semmle/javascript/ApiGraphs.qll:616]: I think that already works, but let me add a test for it.

Comment (max-schaefer) [reply javascript/ql/src/semmle/javascript/ApiGraphs.qll:654]: I have added a test which, to my not inconsiderable surprise, passed.

**Comment by max-schaefer** [comment]
Rebased to take advantage of `getAnExportedValue`. Happy to redo the eval if desired.

**Comment by max-schaefer** [comment]
A [re-evaluation](https://github.com/max-schaefer/dist-compare-reports/blob/59144978ceae91be27ccc2adf3b06c90bcb07470/js/api-graph-reexports/report.md) on big-apps (internal link) suggests a minor slowdown, but given the performance fluctuations I've seen on my last few evaluations I wouldn't read too much into that.

**Review by asgerf — APPROVED** [review]
👍

**Pull request merged**
